redhat-appstudio/service-provider-integration-operator
# Svpi110 repo info

### What does this PR do?
Adds new CR `SPIAccessCheck` with url to the repository, controller checks if repository is public, if user has any tokens for it and write info about repo into `SPIAccessCheck` status. Controller removes CR after some period of time (30mins default, configurable)


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-110


### How to test this PR?

operator image `quay.io/mvala/spi-operator:svpi110-repoInfo`

```
apiVersion: appstudio.redhat.com/v1beta1
kind: SPIAccessCheck
metadata:
  name: spiaccesscheck-sample
spec:
  repoUrl: "<github-repo-url>"
```

create CR ^. 
Try:
 - public repo
 - private repo without existing SPIAccessToken in the namespace
 - private repo with existing SPIAccessToken in the namespace
 - operator should remove SPIAccessCheck CR after 30 minutes (default, configurable with `accessCheckTtl` in config.yaml)
